extension-style/src/main/java/com/mapbox/maps/extension/style/sources/Source.kt

```diff
@@ -93,6 +93,14 @@ abstract class Source(
   private fun updateProperty(property: PropertyValue<*>, throwRuntimeException: Boolean = true) {
     delegate?.let { styleDelegate ->
       try {
+        // checking for validness makes sense only for geojson as it uses async parsing
+        if (this is GeoJsonSource) {
+          // explicitly reset native reference and return if map is not valid anymore
+          if (!styleDelegate.isValid()) {
+            delegate = null
+            return@let
+          }
```

Comment:
Would it make sense to add some error logs here?

Reply:
it's non-error situation - map destroyed while we were parsing so see no need in logs here

Reply:
that code is not correct when using persistent layers, rollbacked in https://github.com/mapbox/mapbox-maps-android/pull/1324